doc/releases/changelog-dev.md

```diff
@@ -163,6 +163,17 @@
    ((0.25+0j)*(PauliX(wires=[0]))) + (0.25j*(PauliY(wires=[0]) @ PauliZ(wires=[1])))
    ```
 
+<h3>Improvements 🛠</h3>
+
+* The `default.qubit` device handles measurement value lists when executing mid-circuit measurements natively.
+  [(#5120)](https://github.com/PennyLaneAI/pennylane/pull/5120)
+
+* The `default.qubit` device handles composite measurements when executing mid-circuit measurements natively.
+  [(#5120)](https://github.com/PennyLaneAI/pennylane/pull/5120)
+
+* The `default.qubit` device handles post-selection when executing mid-circuit measurements natively.
+  [(#5120)](https://github.com/PennyLaneAI/pennylane/pull/5120)
```

Comment:
Let me know what you think about this. I think either is fine though.

```suggestion
* The `default.qubit` device now supports mid-circuit measurements without the use of the deferred measurement
  principle when using finite shots. This includes all features supported with `qml.defer_measurements`, including
  classical control, collecting statistics, and postselection.
  [(#5120)](https://github.com/PennyLaneAI/pennylane/pull/5120)
```

Reply:
Thanks @mudit2812 , read much better for a user. I copy-pasted and added a bit to it.